queries/queries_test.go

```diff
@@ -0,0 +1,176 @@
+package queries_test
+
+import (
+	"fmt"
+	"log"
+	"testing"
+	"time"
+
+	"github.com/jmpsec/osctrl/nodes"
+	"github.com/jmpsec/osctrl/queries"
+	"github.com/stretchr/testify/assert"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+)
+
+func setupTestDB() (*gorm.DB, error) {
+	db, err := gorm.Open(sqlite.Open(":memory:"), &gorm.Config{})
+	if err != nil {
+		return nil, err
+	}
+	return db, nil
+}
+
+func TestNodeQueries(t *testing.T) {
+	db, err := setupTestDB()
+	if err != nil {
+		t.Fatalf("Failed to setup test database: %v", err)
+	}
+
+	// Create tables
+	q := queries.CreateQueries(db)
+	nodes.CreateNodes(db)
+
+	// Create test data
+	node := nodes.OsqueryNode{
+		Model: gorm.Model{ID: 1},
+	}
+	distributedQuery := queries.DistributedQuery{
+		Model:         gorm.Model{ID: 1},
+		Name:          "test_query",
+		Query:         "SELECT * FROM osquery_info;",
+		EnvironmentID: 1,
+		Expiration:    time.Now().Add(24 * time.Hour),
+	}
+	nodeQuery := queries.NodeQuery{
+		NodeID:  1,
+		QueryID: 1,
+		Status:  "pending",
+	}
+
+	// Query sqlite_master to list all tables
+	var tables []string
+	err = db.Raw("SELECT name FROM sqlite_master WHERE type='table'").Scan(&tables).Error
```

Comment:
The backend is postgres, so as long as we don't use anything postgres specific, we can still test the code using sqlite. Otherwise, testing it is a pain due to all the mocking, see [users/users_test.go](https://github.com/jmpsec/osctrl/blob/main/users/users_test.go)

Reply:
Yes, I agree, we can use varchar for now for easy testing.